src/main/scala/com/thirdship/libunit/AStarSolver.scala

```diff
@@ -134,103 +136,86 @@ case class AStarSolver(var allTSUnits: List[String], var allConversions: List[Co
+    *   are stored in the gScores map, keyed by the specified unit.
     *
-    * closedSet is the set of units in the graph that are not the goal. openSet is the set of units that might be the goal.
-    * The openSet, then, represents the fringes of the search, whereas the closedSet represents the insides of the searched map.
+    * closedSet is the set of units in the graph that are not the goal. openSet is the set of units that might be the
+    *   goal. The openSet, then, represents the fringes of the search, whereas the closedSet represents the insides of
+    *   the searched map. The openSet is iterated over until the goal is found, and populated by neighbors of units in
+    *   the openSet.
     *
-    * cameFrom is a map of units to units, storing the value of the unit was visited right before the unit key.
-    * This is used to reconstruct the path taken by the algorithm.
+    * cameFrom is a map of units to units, storing the value of the unit was visited right before the unit key in the
+    *   final path. This is used to reconstruct the path taken by the algorithm. As different ways of traveling from one
+    *   unit to another are found, the path that has the lowest cost is kept, and the higher-cost path is thrown out.
     *
     * @param start  The start unit for the search, the one being converted from.
     * @param end   The goal unit for the search, the one being converted to.
     * @return A conversion from start to goal that is the most conversion-cost-efficient.
     */
   def solve(start: String, end: String): ConversionEdge[String, Double, Double] = {
-    //println("A* algorithm begun! " + start + " is the start and " + goal + " is the goal.")
-    if(!allTSUnits.contains(start)){
-      //TODO replace with exceptions or None
-      val startNotUnit = new ScalarConversionEdge("start is","not a unit!",1)
+    if (! allTSUnits.contains(start)) {
+      val startNotUnit = new ScalarConversionEdge("start is", "not a unit!", 1) // TODO replace with exceptions or None
       return startNotUnit
     }
-    if(!allTSUnits.contains(end)){
-      //TODO replace with exceptions or None
-      val endNotUnit = new ScalarConversionEdge("end is","not a unit!",1)
+    if (! allTSUnits.contains(end)) {
+      val endNotUnit = new ScalarConversionEdge("end is", "not a unit!", 1) // TODO replace with exceptions or None
       return endNotUnit
     }
     var closedSet = List.empty[String]
     var openSet = List.empty[String]
     openSet :+= start
-    var cameFrom = Map.empty[String,String]
-    // All the units populate cameFrom as keys, and are adjusted as the algorithm finds paths.
-    var current: String = null
+    val cameFrom = mutable.Map.empty[String, String]
+    var current: String = ""
 
-    var gScore = Map.empty[String,Double]
-    var fScore = Map.empty[String,Double]
+    val gScore = mutable.Map.empty[String, Double]
+    val fScore = mutable.Map.empty[String, Double]
```

Comment:
I would rather the pointer be a var than the map be mutable. The map being mutable means that we could run into concurrency issues. With a mutable pointer we could risk mutations not taking effect immediately, but this is better than the alt.

What was the reason for the change?